src/main/java/com/aws/greengrass/lifecyclemanager/GenericExternalService.java

```diff
@@ -698,19 +730,28 @@ protected RunResult run(String name, Topic t, String cmd, IntConsumer background
         exec = addShell(exec);
 
         addEnv(exec, t.parent);
-        logger.atDebug().setEventType("generic-service-run").log();
 
-        // Track all running processes that we fork
-        if (exec.isRunning()) {
-            trackingList.add(exec);
+        Exec finalExec = exec;
+        CrashableSupplier<RunStatus, InterruptedException> doRun = () -> {
+            logger.atDebug().setEventType("generic-service-run").log();
+            // Track all running processes that we fork
+            if (finalExec.isRunning()) {
+                trackingList.add(finalExec);
+            }
+            return shellRunner.successful(finalExec, t.getFullName(),
+                    background, this) ? RunStatus.OK : RunStatus.Errored;
```

Comment:
When a cached exec runs and fails, looks like it won't update the status code in `RunResult`?

Reply:
The runresult isn't used by anyone, so not, it isn't updated, but it doesn't matter.

Reply:
Why is it not used? if shutdown fails we need to set the status code correctly no?

Reply:
also the `RunStatus` is still returned by the cached method, so you can check the result if you wanted.

Reply:
This is our current implementation: https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/lifecyclemanager/GenericExternalService.java#L536-L540

Reply:
RunStatus doesn't have enough info, but since it's not used right now and component doesn't go to errored state this is probably okay